app/src/main/java/com/termux/app/ExtraKeysInfos.java

```diff
@@ -0,0 +1,222 @@
+package com.termux.app;
+
+import androidx.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ExtraKeysInfos {
+
+    /**
+     * Matrix of buttons displayed
+     */
+    private String [][][] buttons;
+
+    /**
+     * This corresponds to one of the CharMapDisplay below
+     */
+    private String style = "default";
+
+    /**
+     * This is the keyMap defined in termux.properties
+     */
+    Map<String, String> keyMap;
+
+    /**
+     * HashMap that implements Python dict.get(key, default) function.
+     * Default java.util .get(key) is then the same as .get(key, null);
+     */
+    static class CleverMap<K,V> extends HashMap<K,V> {
```

Comment:
Java 1.8 (which is the version used as far as I can see) has `getOrDefault`.

Reply:
Oh, sorry, I saw now that this wasn't something new, just moved from the other class.

Reply:
Yes, I began by creating a separate class to hold the data of extra-keys which will become more complex.